src/native/fs/fs_napi.cpp

```diff
@@ -213,6 +214,112 @@ struct Stat : public FSWorker
     }
 };
 
+/**
+ * LStat is an fs op
+ */
+struct LStat : public FSWorker
```

Comment:
LStat and Stat are identical.
Except for the symbolic link which is transparent to us since we just call the POSIX and the FS handles that.
Maybe instead of copy/pasting the code and maintaining similar copies to code, it is better to provide an option to the native layer and call the functions respectively?
Maybe even having the fstat as well passing an FD as an option.
These three functions basically have the same code and the only difference is what POSIX we are calling.
You can check the first parameter if number (fd) or string (path) and that way you between fstat and regular stat, and the second parameter will be provided when there is a requirement to do lstat.

Reply:
Good point. I agree that the similarity is high.
But I would prefer to see the code reuse inside and still allow different function names outside to match the well known linux syscalls.